pkg/controller/migration/namespace_migration.go

```diff
@@ -220,18 +220,22 @@ func (m *CoreNamespaceMigration) Run(ctx context.Context, log logr.Logger) error
 		return fmt.Errorf("failed deleting kube-system calico-kube-controllers: %s", err.Error())
 	}
 	log.V(1).Info("Deleted previous calico-kube-controllers deployment")
-	if err := m.waitForOperatorTyphaDeploymentReady(ctx); err != nil {
-		return fmt.Errorf("failed to wait for operator typha deployment to be ready: %s", err.Error())
-	}
-	log.V(1).Info("Operator Typha Deployment is ready")
 	if err := m.labelUnmigratedNodes(ctx); err != nil {
 		return fmt.Errorf("failed to label unmigrated nodes: %s", err.Error())
 	}
 	log.V(1).Info("All unmigrated nodes labeled")
-	if err := m.ensureKubeSysNodeDaemonSetHasNodeSelectorAndIsReady(ctx); err != nil {
+	if err := m.ensureKubeSysNodeDaemonSetHasNodeSelectorAndIsReady(ctx, log); err != nil {
 		return fmt.Errorf("the kube-system node DaemonSet is not ready with the updated nodeSelector: %s", err.Error())
 	}
 	log.V(1).Info("Node selector added to kube-system node DaemonSet")
+	if err := m.ensureTyphaRoom(ctx, log); err != nil {
```

Comment:
I still need to delete / disable any auto-scalers running against the old cluster before calling this function.